src/test/java/seedu/clinic/logic/commands/DeleteCommandTest.java

```diff
@@ -79,6 +103,19 @@ public void execute_invalidIndexUnfilteredList_throwsCommandException() {
         assertCommandFailure(deleteCommand, modelForWarehouse, Messages.MESSAGE_INVALID_WAREHOUSE_DISPLAYED_INDEX);
     }
 
+    @Test
+    public void execute_validIndexInvalidProductNameDeletion_throwsCommandException() {
+        Supplier supplierToUpdate = modelForSupplier.getFilteredSupplierList().get(INDEX_FIRST_SUPPLIER.getZeroBased());
+        Name invalidProductToDeleteName = new Name(VALID_PRODUCT_NAME_ASPIRIN);
+
+        String expectedMessage = String.format(Messages.MESSAGE_INVALID_PRODUCT_NAME_IN_SUPPLIER,
+                invalidProductToDeleteName, supplierToUpdate.getName());
+        DeleteCommand deleteCommand =
+                new DeleteCommand(TYPE_SUPPLIER, INDEX_FIRST_SUPPLIER, invalidProductToDeleteName);
+
+        assertCommandFailure(deleteCommand, modelForSupplier, expectedMessage);
+    }
+
```

Comment:
Do we have tests for invalid index when deleting products??

Reply:
I think this is covered in the delete supplier/warehouse part, I can add this test again later if needed, thanks for the comment!

Reply:
Ahh ok! Thanks for the update!!